ale_linters/javascript/flow.vim

```diff
@@ -71,36 +74,70 @@ function! ale_linters#javascript#flow#Handle(buffer, lines) abort
         let l:text = ''
         let l:line = 0
         let l:col = 0
-
-        for l:message in l:error.message
-            " Comments have no line of column information, so we skip them.
-            " In certain cases, `l:message.loc.source` points to a different path
-            " than the buffer one, thus we skip this loc information too.
-            if has_key(l:message, 'loc')
-            \&& l:line is# 0
-            \&& ale#path#IsBufferPath(a:buffer, l:message.loc.source)
-                let l:line = l:message.loc.start.line + 0
-                let l:col = l:message.loc.start.column + 0
-            endif
-
-            if l:text is# ''
-                let l:text = l:message.descr . ':'
-            else
-                let l:text = l:text . ' ' . l:message.descr
-            endif
-        endfor
+        let l:detail = ''
+
+        if has_key(l:error, 'message')
+            for l:message in l:error.message
+                " Comments have no line of column information, so we skip them.
+                " In certain cases, `l:message.loc.source` points to a different path
+                " than the buffer one, thus we skip this loc information too.
+                if has_key(l:message, 'loc')
+                \&& l:line is# 0
+                \&& ale#path#IsBufferPath(a:buffer, l:message.loc.source)
+                    let l:line = l:message.loc.start.line + 0
+                    let l:col = l:message.loc.start.column + 0
+                endif
+
+                if l:text is# ''
+                    let l:text = l:message.descr . ':'
+                else
+                    let l:text = l:text . ' ' . l:message.descr
+                endif
+            endfor
+        endif
 
         if has_key(l:error, 'operation')
             let l:text = l:text . ' See also: ' . l:error.operation.descr
         endif
 
+        if has_key(l:error, 'extra')
```

Comment:
Could you move this part to another function and call it here? You could call it `s:GetDetail` or something similar.